frontend/packages/dev-console/src/components/pipelines/utils/pipeline-operator.ts

```diff
@@ -0,0 +1,25 @@
+import { compare, parse, SemVer } from 'semver';
+import {
+  ClusterServiceVersionKind,
+  ClusterServiceVersionModel,
+  ClusterServiceVersionPhase,
+} from '@console/operator-lifecycle-manager';
+import { k8sList } from '@console/internal/module/k8s';
+
+export const getPipelineOperatorVersion = async (namespace: string): Promise<SemVer | null> => {
+  const allCSVs: ClusterServiceVersionKind[] = await k8sList(ClusterServiceVersionModel, {
+    ns: namespace,
+  });
+  const matchingCSVs = allCSVs.filter(
+    (csv) =>
+      csv.metadata?.name?.startsWith('openshift-pipelines-operator') &&
+      csv.status?.phase === ClusterServiceVersionPhase.CSVPhaseSucceeded,
+  );
+  const versions = matchingCSVs.map((csv) => parse(csv.spec.version)).filter(Boolean);
+  // Orders from small (oldest) to highest (newest) version
```

Comment:
I'm confused by this comment vs the tests.

```suggestion
  // Orders versions from latest to earliest
```
You're grabbing the last one, the earliest version based on your tests.

Reply:
The order after this `versions.sort(compare)` is 1.0, 1.1, 2.0, ... And then I selected the last one (to get the newest version). The comment was only for the `versions.sort` line. I will update that comment so that it matches both lines or do some code change. don't know yet.

Reply:
Ah I was confused by your test... I didn't read the `phase` value. You're right. Apologizes -- ignore this thread.